nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/test/groovy/org/apache/nifi/properties/AESSensitivePropertyProviderTest.groovy

```diff
@@ -86,7 +86,7 @@ class AESSensitivePropertyProviderTest extends GroovyTestCase {
                 if (Cipher.getMaxAllowedKeyLength("AES") < keySize) {
                     throw new IllegalArgumentException("The JCE unlimited strength cryptographic jurisdiction policies are not installed, so the max key size is 128 bits")
                 }
-                return KEY_256_HEX[0..<(keySize / 4)]
+                return KEY_256_HEX[0..<keySize.intdiv(4)]
```

Comment:
Curious what the difference is here. When I changed to 2.5.7 I remember the original line returning 49 instead of 48.

Reply:
A result of `BigDecimal` is returned by this division.  Even though, when using the debugger, the result of the division is a `BigDecimal` with a value of 48, the resulting array has a length of 49.  I didn't dig much deeper than that.  Using `intdiv` returns a `Number` with a value of 48, but the array is created with a length of 48, which is the correct size.  It must boil down to a change in how Groovy 2.4.x and Groovy 2.5.x use the type of the values used for the bounds of the slice of the array.  It's still unclear to me why a `BigDecimal` with a value of 48 results in the array being created with a length of 49.